deepchem/models/tensorgraph/layers.py

```diff
@@ -140,6 +140,129 @@ def convert_to_layers(in_layers):
   return layers
 
 
+class AlphaShare(Layer):
+    """
+    Part of a sluice network. Adds alpha parameters to control
+    sharing between the main and auxillary tasks
+    """
+
+    def __init__(self, **kwargs):
+        super(AlphaShare, self).__init__(**kwargs)
+
+    def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+        inputs = self._get_input_tensors(in_layers)
+        # check that there isnt just one or zero inputs
+        if len(inputs) <= 1:
+            raise ValueError("AlphaShare must have more than one input")
+
+        # create subspaces
+        subspaces = []
+        original_cols = int(inputs[0].get_shape()[-1].value)
+        subspace_size = int(original_cols / 2)
+        for input_tensor in inputs:
+            subspaces.append(tf.reshape(input_tensor[:, :subspace_size], [-1]))
+            subspaces.append(tf.reshape(input_tensor[:, subspace_size:], [-1]))
+        n_alphas = len(inputs) * 2
+        subspaces = tf.reshape(tf.stack(subspaces), [n_alphas, -1])
+
+        # create the alpha learnable parameters
+        alphas = tf.Variable(tf.random_normal(
+            [n_alphas, n_alphas]), name='alphas')
+
+        out_tensor = tf.matmul(alphas, subspaces)
+
+        # concatenate subspaces, reshape to size of original input, then stack
+        # such that out_tensor has shape (2,?,original_cols)
+        row = 0
+        lin_comb = []
+        for x in range(0, len(inputs)):
+            lin_comb.append(tf.reshape(
+                out_tensor[row:row + 2, ], [-1, original_cols]))
+            row += 2
+        out_tensor = tf.stack(lin_comb)
+        self.alphas = alphas
+        if set_tensors:
+            self.out_tensor = out_tensor
+        return out_tensor
+
```

Comment:
p.s. you also need to implement none_tensors and set_tensors if you set class variables that are tensor objects.

If you don't these layers will not be save-able